provider/provider.go

```diff
@@ -164,24 +164,26 @@ func (clientTLS *ClientTLS) CreateTLSConfig() (*tls.Config, error) {
 	cert := tls.Certificate{}
 	_, errKeyIsFile := os.Stat(clientTLS.Key)
 
-	if _, errCertIsFile := os.Stat(clientTLS.Cert); errCertIsFile == nil {
-		if errKeyIsFile == nil {
-			cert, err = tls.LoadX509KeyPair(clientTLS.Cert, clientTLS.Key)
-			if err != nil {
-				return nil, fmt.Errorf("Failed to load TLS keypair: %v", err)
+	if len(clientTLS.Cert) > 0 && len(clientTLS.Key) > 0 {
```

Comment:
The problem is that `CreateTLSConfig()` is being used from many places, where some are using this for outbound connections, for example Marathon provider uses this to create an http client to Mesos Master like:
```TLSConfig, err := p.TLS.CreateTLSConfig()
		if err != nil {
			return err
		}
		config.HTTPClient = &http.Client{
			Transport: &http.Transport{
				DialContext: (&net.Dialer{
					KeepAlive: time.Duration(p.KeepAlive),
					Timeout:   time.Duration(p.DialerTimeout),
				}).DialContext,
				TLSClientConfig: TLSConfig,
			},
		}
		client, err := marathon.NewClient(config)```
If you have Mesos Master using self-signed certificate, then all you need to configure is `InsecureSkipVerify=true`, but you don't have either a cert or a key to provide.  But there are other cases where you would have either a cert or a key, as well as `InsecureSkipVerify=true`. (but let me know if you think the above use-case is not accurate), I'd change the condition to just check for `if ! clientTLS.InsecureSkipVerify` instead.